cmd/activator/main.go

```diff
@@ -145,8 +145,19 @@ func main() {
 		logger.Fatalw("Error building serving clientset", zap.Error(err))
 	}
 
-	if err := version.CheckMinimumVersion(kubeClient.Discovery()); err != nil {
-		logger.Fatalf("Version check failed: %v", err)
+	// We sometimes startup faster than we can reach kube-api. Poll on failure to prevent us terminating
+	// our pod
+	startTime := time.Now()
+	for {
+		if err := version.CheckMinimumVersion(kubeClient.Discovery()); err != nil {
+			logger.Errorf("Version check failed: %v", err)
```

Comment:
For zap logger use `Errorw` as it breaks out the error into a structured searchable field

```suggestion
			logger.Errorw("Version check failed", zap.Error(err))
```